### PR TITLE
added email verification

### PR DESCRIPTION
**Pull request by 4p00rv**
Added functionality to verify email addresses of member.

### CODE REVIEW

**Comment by 4p00rv** [comment]
Had to create a new pull request. Still confused with how Github works. Please review.

File: app/controllers/members_controller.rb [hunk 1]
```diff
@@ -24,18 +26,38 @@ def create
...
+      MemberMailer.welcome_email(@member).deliver_now
+      flash[:success] = "Welcome to the S'Up! Your email has been confirmed."
+      redirect_to action: :index
+    else
+      flash[:error] = 'Invalid token'
+      redirect_to action: :index
+    end
+  end
+
+  def getter
```
Comment (dblock) [inline app/controllers/members_controller.rb]: `getter` => `get_member`, also a Rails convention.

File: app/controllers/members_controller.rb [hunk 2]
```diff
@@ -24,18 +26,38 @@ def create
...
     @member = Member.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    not_found
+  end
+
+  def not_found
+    fail ActionController::RoutingError, 'Mr. Holmes has been notified about the missing member'
```
Comment (dblock) [inline app/controllers/members_controller.rb]: :+1:

File: app/controllers/members_controller.rb [hunk 3]
```diff
@@ -62,6 +84,10 @@ def graph
     @link_data_str = edges.to_json.html_safe
   end
 
+  def check_email_confirmed?
```
Comment (dblock) [inline app/controllers/members_controller.rb]: That's not the convention, when you want to say `member.present?` you use methods with question marks, so `check_email_confirmed`.

**Comment by dblock** [comment]
Good, fix the names of methods via `--amend`, and @ilyakava feel free to review/merge.

File: spec/controllers/members_controller_spec.rb [hunk 4]
```diff
@@ -38,4 +38,48 @@
...
+    end
+
+    it 'checks member who have not verified email yet does not have access' do
+      get :edit, id: @member
+      expect(response.status).to eq 302
+    end
+  end
+
+  it 'redirects to 404 when a member is not found' do
+    random_member_id = (2 + rand(10)).to_s
```
Comment (dblock) [inline spec/controllers/members_controller_spec.rb]: This makes the test unpredictable and does unnecessary work. Once in a  blue moon you'll get the same id as an actual member. Very simply just pass `id: 'invalid'`, or if this needs to be a number `id: -1` for example.

File: spec/models/meeting_spec.rb [hunk 5]
```diff
@@ -78,7 +78,7 @@ def make_groups_and_members!(groups, distributions)
   end
   describe 'a perfectly symmetrical example' do
     before :each do
-      groups = %w(      A B C      )
+      groups = %w( A B C      )
```
Comment (dblock) [inline spec/models/meeting_spec.rb]: What's up with the trailing space after the C? :)

**Comment by dblock** [comment]
The `Gemfile.lock` change should be reverted, we do like gems upgraded but that's not related to this change.

File: Rakefile [hunk 6]
```diff
@@ -3,7 +3,9 @@
 
 require File.expand_path('../config/application', __FILE__)
 require 'rspec/core/rake_task'
+require 'rubocop'
```
Comment (dblock) [inline Rakefile]: If you're going to require `rubocop` then you can do:

```ruby
require 'rubocop/rake_task'
RuboCop::RakeTask.new(:rubocop)
```

Again, unrelated change should go into a separate PR.

File: spec/models/meeting_spec.rb (hunk 5 above)
Comment (4p00rv) [inline spec/models/meeting_spec.rb]: Something's really wrong with rubocop, I tried `rubocop -a`, but it did the same! Might be a bug in rubocop

File: app/controllers/members_controller.rb (hunk 1 above)
Comment (4p00rv) [inline app/controllers/members_controller.rb]: it raises a rubocop offense!

**Comment by 4p00rv** [comment]
Waiting for @ilyakava to review.

**Comment by dblock** [comment]
:+1: 

@4p00rv What do we need to run on a console to trigger e-mail verification for existing users?

**Comment by 4p00rv** [comment]
I missed that in the migration, should I create a new migration or add the code to update existing users in the same migration?

**Comment by dblock** [comment]
I think it's a good idea @4p00rv.

**Comment by 4p00rv** [comment]
Should I add a new migration?

**Comment by dblock** [comment]
Not sure. Try something and we'll take a look.

**Comment by 4p00rv** [comment]
I have added a new migration. Please take a look

**Comment by dblock** [comment]
I think it's good, this way any existing user is verified after a migration.

@ilyakava all yours

**Comment by ilyakava** [comment]
kk, very slick @4p00rv, thank you for all the work! @dblock covered everything. :boom:

**Comment by 4p00rv** [comment]
:+1:  Happy to be of help :)
